### PR TITLE
[Enhancement][BugFix] fix the problem that the spill dir cannot be deleted when the query exits abnormally

### PR DESCRIPTION
**Pull request by silverbullet233**
1. fix the problem that the spill directory cannot be deleted when the query exits abnormally
2. add more metrics to observe the memory occupied by buffers in each operator, such as the chunk buffer of scan operator, the passthrough buffer of exchange operator, etc.
3. add more metrics to observe the number of spill io tasks
4. add more metrics to observe the time spent in spill mem table sort

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/exec/spill/log_block_manager.cpp [hunk 1]
```diff
@@ -45,6 +45,8 @@ class LogBlockContainer {
     ~LogBlockContainer() {
         TRACE_SPILL_LOG << "delete spill container file: " << path();
         WARN_IF_ERROR(_dir->fs()->delete_file(path()), fmt::format("cannot delete spill container file: {}", path()));
+        // try to delete related dir, only the last one can success, we ignore the error
+        (void)(_dir->fs()->delete_dir(parent_path()));
```
Comment (stdpain) [inline be/src/exec/spill/log_block_manager.cpp:49]: this logical should be in DirManager?

**Review by fzhedu — APPROVED** [review]

File: be/src/exec/spill/log_block_manager.cpp (hunk 1 above)
Comment (silverbullet233) [reply be/src/exec/spill/log_block_manager.cpp:49]: > this logical should be in DirManager?

nope, DirManager does not care about specific queries, it is only responsible for allocating writable directories.
LogBlockContainer is bound to the query, we need to handle the creation and deletion of the directory during its life cycle

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**